### ISSUE TITLE
Question about padding_mask

### ISSUE THREAD

**Issue by kayden-kim0702** [ddlBoJack/emotion2vec#63]
i am trying to using frame-level latent feature and i faced some problem about padding_mask.

My code is as follow:

import fairseq
from dataclasses import dataclass
@dataclass
class UserDirModule:
    user_dir: str

model_dir='./emotion2vec/upstream'
checkpoint_dir='emotion2vec_base.pt'
model_path = UserDirModule(model_dir)
fairseq.utils.import_user_module(model_path)
emovec_model, cfg, task = fairseq.checkpoint_utils.load_model_ensemble_and_task([checkpoint_dir])
emovec_model = emovec_model[0]
emovec_model=emovec_model.eval().to(device)

### This is a single instance example (not a batch inference)

emovec = emovec_model.extract_features(audio, padding_mask=pad_mask)
emovec = emovec['x'].transpose(2,1).detach().squeeze(0).detach().cpu()

emovec2 = emovec_model.extract_features(audio, padding_mask=None)
emovec2 = emovec2['x'].transpose(2,1).detach().squeeze(0).detach().cpu()

The output results of emovec and emovec2 are quit different. 
I may guess that emovec is correct result but i am not sure why their results are  different. 

***** Update
When i check the attention mask, Final frame (time step) is ignored during the attention computation process.  As a result, attention make will be 
attention_mask=[False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False, False, False, False, False, False, False, False,
         False, False, False,  True]]

and final frame is ignored. it make the different between  emovec and emovec2.

From this point of view, should i ignore final time step after extracting latent feature??

ex)
emovec=emovec[:,:emovec.size(1)-1].   ignore final time step

**Comment by ddlBoJack** [comment]
Hi, please set `padding_mask=None` for feature extraction.

Refer to [this page](https://github.com/ddlBoJack/emotion2vec/blob/main/scripts/extract_features.py)

**Comment by kayden-kim0702** [comment]
Thank you to give a good response. 
If i set padding_mask  = None, how can i conduct batch inference??
